packages/fe-mockserver-core/src/router/batchRouter.ts

```diff
@@ -63,21 +190,29 @@ export function batchRouter(dataAccess: DataAccess): NextHandleFunction {
             const batchData = parseBatch(new BatchContent(body), boundary);
             const globalHeaders: Record<string, string> = {};
             let batchResponse = '';
-
+            //initialize the instance of aggregator of batch 412 responses
+            aggregate412BatchResponseInstance = aggregate412BatchResponse();
             for (const part of batchData.parts) {
                 if (isPartChangeSet(part)) {
                     batchResponse += `--${batchData.boundary}${NL}`;
                     batchResponse += `Content-Type: multipart/mixed; boundary=${part.boundary}${NL}`;
                     batchResponse += NL;
                     for (const changeSetPart of part.parts) {
-                        batchResponse += await handlePart(
+                        const batchPartRes = await handlePart(
                             changeSetPart as BatchPart,
                             dataAccess,
                             part.boundary,
                             req.tenantId!,
-                            globalHeaders
+                            globalHeaders,
+                            true
                         );
+                        if (batchPartRes !== undefined) {
+                            // may be make it null in case of 412
```

Comment:
Maybe or maybe not ?

Reply:
removed comment and changed to null comparison